index.js

```diff
@@ -163,13 +164,20 @@ StreamingCache.prototype.set = function (key) {
     emitters[key] = new EventEmitter();
```

Comment:
This function is a bit long. We can extract some methods. stream._read and stream._write can be moved in different functions